pkg/chunk/aws/s3_storage_client.go

```diff
@@ -94,7 +98,12 @@ type S3ObjectClient struct {
 
 // NewS3ObjectClient makes a new S3-backed ObjectClient.
 func NewS3ObjectClient(cfg S3Config, delimiter string) (*S3ObjectClient, error) {
-	s3Config, bucketNames, err := buildS3Config(cfg)
+	return NewS3ObjectClientWithMiddleware(cfg, delimiter, nil)
+}
+
+// NewS3ObjectClientWithMiddleware makes a new S3-backed ObjectClient with an option to add request middleware
+func NewS3ObjectClientWithMiddleware(cfg S3Config, delimiter string, injectMiddleware InjectRequestMiddleware) (*S3ObjectClient, error) {
```

Comment:
**suggestion**: I think we should avoid adding `InjectRequestMiddleware` as a parameter to this function call. Instead we could make it a field in the config.

We can add it to the s3 config with a nil struct tag like:

```yaml
Inject InjectRequestMiddleware `yaml:"-"`
```

When `NewS3ObjectClient` is called, if the config value is set to nil it will use the default behavior. Otherwise, it will apply the middleware. 

**WDYT**?

Reply:
Works for me.  Requested changes pushed.